ivr/src/main/java/org/motechproject/ivr/service/impl/ConfigServiceImpl.java

```diff
@@ -52,10 +52,11 @@ private synchronized void loadConfigs() {
             throw new JsonIOException(message, e);
         }
 
-        configs = new HashMap<>();
-        for (Config config : configList) {
-            configs.put(config.getName(), config);
+        for (Config config : configs.getConfigList()) {
```

Comment:
You should create new map. It will contains old configs, for example if it will contains key_1 and when you add new configs and remowe old(key_1) in the UI then this map will still contain key_1 config.